### PR TITLE
Fixes for FR policy fixing up SELinux context of ~cftransport/.ssh

### PR DESCRIPTION
**Pull request by vpodzime**
Multiple issues discovered and fixed. Plus one piece of polishing.

### CODE REVIEW

**Comment by vpodzime** [comment]
Manually tested to verify that it fixes issues on a CentOS 7 hub enabled as a feeder. Really strange it hasn't been breaking FR tests.

**Comment by vpodzime** [comment]
> Manually tested to verify that it fixes issues on a CentOS 7 hub enabled as a feeder.

It just requires two agent runs to properly do the full setup. https://github.com/cfengine/masterfiles/pull/2817 adds a more complicated change making sure a single agent run is enough, but we may want to stick to this simpler change for backports.

File: cfe_internal/enterprise/federation/federation.cf [hunk 1]
```diff
@@ -300,29 +300,34 @@ bundle agent transport_user
           expression => not(returnszero("$(default:paths.semanage) fcontext -l | grep '$(home)/.ssh(/.*)?'", "useshell")),
           if => fileexists("$(home)");
     enabled.selinux_enabled::
+        # For all the files below it must be true that if they exist they need
```
Comment (craigcomstock) [inline cfe_internal/enterprise/federation/federation.cf:303]: I appreciate your explanation here but have to say this block has become rather unreadable. I wonder if we could do better? The logic reads to me as correct after some time looking at it.

**Review by craigcomstock — APPROVED** [review]

File: cfe_internal/enterprise/federation/federation.cf (hunk 1 above)
Comment (vpodzime) [reply cfe_internal/enterprise/federation/federation.cf:303]: Well, I know how to make this a lot nicer in Python. Or any language with functions and loops, really. But no clue how to make it nicer in CFEngine policy. Maybe @nickanderson can help polish this later.

Comment (vpodzime) [reply cfe_internal/enterprise/federation/federation.cf:303]: We could use a list of files and then create an array with those files as keys and the results of the SELinux labels as values and then search for a false value in `getindices()` on the array. But that's similarly ugly, AFAICT.

**Pull request merged**